scripts/helpers/doxygen.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+CHIP_ROOT="$(dirname "$0")/../.."
+CHIP_NAME="Connected Home over IP"
+CHIP_VERSION="$(git rev-parse --short HEAD)"
+
+die() {
+    echo " *** ERROR: $*"
+    exit 1
+}
+
+install_doxygen() {
+    if ! doxygen --version >/dev/null 2>&1; then
+        echo "doxygen not installed. Installing..."
+        case "$(uname)" in
+            "Darwin")
+                brew install doxygen
+                ;;
+            "Linux")
+                apt-get update
+                apt-get install -y doxygen
```

Comment:
I don't think this is going to work for Linux users, because they won't be building documentation with root privileges.

Reply:
This part comes from #2703. Do you want me to use `sudo`?

Reply:
Sure, or perhaps just print a message of what to run for the user to copy/paste. We need to be able to build stuff in environment without `sudo` access too.